### PR TITLE
Feat/485 further improvements avg speed

### PR DESCRIPTION
**Pull request by ArdtK**
## Issue addressed
Solves #<issue number>

## Code of conduct
- [x] I HAVE NOT added sensitive or compromised (test) data to the repository.
- [x] I HAVE NOT added vulnerabilities to the repository.
- [ ] I HAVE discussed my solution with (other) members of the RA2CE team.

## What has been done?
Explain how you addressed the resolution of the related issue, what choices you made and why.

### Checklist
- [x] Code is formatted using our custom `black` and `isort` definitions.
- [x] Tests are either added or updated.
- [ ] Branch is up to date with `master`.
- [x] Updated documentation if needed.

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

**Review by Carsopre — CHANGES_REQUESTED** [review]
In general no extraordinary remarks, yet I do not  understand the functionality or the tests from `length_weighing_analysis.extend_graph`.

File: tests/network/avg_speed/test_avg_speed_reader.py [hunk 1]
```diff
@@ -15,18 +15,14 @@ def test_initialize(self):
...
-        _expected_values = [
-            [[RoadTypeEnum.TERTIARY], 58.4],
-            [[RoadTypeEnum.TERTIARY_LINK], 58.4],
-            [[RoadTypeEnum.SECONDARY], 0.0],
-            [[RoadTypeEnum.SECONDARY_LINK], 0.0],
-            [[RoadTypeEnum.TRUNK], 0.0],
-            [[RoadTypeEnum.TRUNK_LINK], 0.0],
-            [[RoadTypeEnum.TERTIARY, RoadTypeEnum.SECONDARY], 60.0],
-        ]
+        _expected_values = avg_speed_data
```
[the remark this answers is not in the document]
Comment (ArdtK) [reply tests/network/avg_speed/test_avg_speed_reader.py:25]: Agree

File: tests/network/avg_speed/conftest.py [hunk 2]
```diff
@@ -3,10 +3,24 @@
...
+@pytest.fixture(name="avg_speed_data")
+def get_avg_speed_data() -> Iterator[list[tuple[list[RoadTypeEnum], float]]]:
+    yield [
+        ([RoadTypeEnum.TERTIARY], 1.2),
+        ([RoadTypeEnum.TERTIARY_LINK], 2.3),
+        ([RoadTypeEnum.SECONDARY], 3.4),
+        ([RoadTypeEnum.SECONDARY_LINK], 4.5),
+        ([RoadTypeEnum.TRUNK], 5.6),
+        ([RoadTypeEnum.TRUNK_LINK], 6.7),
+        ([RoadTypeEnum.TERTIARY, RoadTypeEnum.SECONDARY], 7.8),
```
[the remark this answers is not in the document]
Comment (ArdtK) [reply tests/network/avg_speed/conftest.py:25]: Done

File: tests/analysis/losses/weighing_analysis/test_time_weighing_analysis.py [hunk 3]
```diff
@@ -60,3 +60,17 @@ def test_calculate_alternative_value(self, valid_analysis: TimeWeighingAnalysis)
...
         assert _calculated_time == pytest.approx(_expected_time, rel=1e-2)
+
+    def test_extend_graph(self, valid_analysis: TimeWeighingAnalysis):
+        # 1. Define test data.
+        _graph_dict = dict(my_value=42)
+        _time_list = [4.2]
+        valid_analysis.time_list = _time_list
+
+        # 2. Run test.
+        valid_analysis.extend_graph(_graph_dict)
```
[the remark this answers is not in the document]
Comment (ArdtK) [reply tests/analysis/losses/weighing_analysis/test_time_weighing_analysis.py:71]: Actually, this is your code I restored ;-). In general I agree.

File: tests/analysis/losses/weighing_analysis/test_length_weighing_analysis.py [hunk 4]
```diff
@@ -60,3 +59,14 @@ def test_calculate_alternative_value(self, valid_analysis: LengthWeighingAnalysi
...
+    def test_extend_graph(self, valid_analysis: LengthWeighingAnalysis):
+        # 1. Define test data.
+        _graph_dict = dict(my_value=42)
+
+        # 2. Run test.
+        valid_analysis.extend_graph(_graph_dict)
+
+        # 3. Verify expectations
+        assert len(_graph_dict) == 1
+        assert _graph_dict["my_value"] == 42
```
[the remark this answers is not in the document]
Comment (ArdtK) [reply tests/analysis/losses/weighing_analysis/test_length_weighing_analysis.py:72]: See previous remark.

File: ra2ce/analysis/losses/weighing_analysis/weighing_analysis_protocol.py [hunk 5]
```diff
@@ -24,3 +26,8 @@ def calculate_alternative_value(self, alt_dist: float) -> float:
         Returns:
             float: Corrected alternative distance/time value.
         """
+
+    def extend_graph(self, gdf_graph: gpd.GeoDataFrame | dict) -> None:
+        """
+        Extends the provided graph with custom attributes.
+        """
```
[the remark this answers is not in the document]
Comment (ArdtK) [reply ra2ce/analysis/losses/weighing_analysis/weighing_analysis_protocol.py:33]: Let's have a chat on the weighing calculator in general.

File: ra2ce/analysis/losses/weighing_analysis/length_weighing_analysis.py [hunk 6]
```diff
@@ -37,3 +29,6 @@ def get_current_value(self) -> float:
 
     def calculate_alternative_value(self, alt_dist: float) -> float:
         return alt_dist
+
+    def extend_graph(self, gdf_graph: gpd.GeoDataFrame | dict) -> None:
+        return
```
[the remark this answers is not in the document]
Comment (ArdtK) [reply ra2ce/analysis/losses/weighing_analysis/length_weighing_analysis.py:34]: Same comment

**Pull request merged**